best_practices/source_control/terra_source_control_cheatsheet.md

```diff
@@ -0,0 +1,256 @@
+    - [(2) Use the terminal to set up and clone your Git repository](#2-use-the-terminal-to-set-up-and-clone-your-git-repository)
+    - [(3) Check in your code changes](#3-check-in-your-code-changes)
+  - [FAQ](#faq)
+    - [How do I make changes to my branch?](#how-do-i-make-changes-to-my-branch)
+    - [I accidentally checked a new change to my main branch instead of a feature branch. How to fix this?](#i-accidentally-checked-a-new-change-to-my-main-branch-instead-of-a-feature-branch-how-to-fix-this)
+      - [How do I revert a modified file to its committed version?](#how-do-i-revert-a-modified-file-to-its-committed-version)
+
+This document summarizes some of the key practices described by the three-part document in this directory ([Part I](./terra_source_control_I.md), [Part II](./terra_source_control_II.md), [Part III](./terra_source_control_III.md)). It covers both [Terra.bio](https://app.terra.bio/) (Terra) workspaces and **All of Us Workbench** (AoU) workspaces.  In some cases there are differences between the two platforms.
+
+<!--It's primarily intended for analysts using source control for the materials within the featured workspaces, but these source control practices could be used by any AoU researcher who would like to use source control to manage their analysis code.-->
+
+## Check only code (no data) into source control
+
+### Exclude notebook outputs
+
+For the **All of Us Workbench**, the most important thing to remember is **notebooks checked into source control may not contain outputs**. Use [nbstripout](https://github.com/kynan/nbstripout) to remove all notebook outputs.
+This screencast demonstrates the use and working principles behind the `nbstripout` utility and how to use it as a Git filter:
+
+[![Img alt text](https://i.imgur.com/7oQHuJ5.png)](https://www.youtube.com/watch?v=BEMP4xacrVc)
+
+On the **All of Us Workbench**, if you are using Git from the terminal of your workspace,
+`nbstripout` is preinstalled and enabled. You can run `nbstripout --status` to confirm this.
+
+For Terra example workspaces, you may decide to retain some notebook outputs for explanatory value,
+depending upon context.  Be sure that you don't retain any sensitive data or personal config.
+
+
+### Exclude data files
+
+For your repository, use a [gitignore](https://git-scm.com/docs/gitignore) file similar to the
+[.gitignore](https://github.com/DataBiosphere/terra-example-notebooks/blob/main/.gitignore) in this
+repository to ensure that CSVs and image files are not accidentally commited to the repository.
+
+If you are using Git from the terminal of your AoU workbench machine, a global `gitignore` file is
+preinstalled and enabled. You can view file `/home/jupyter/gitignore_global` to confirm this.
+
+On Terra, depending upon context, you may want to make some exceptions to this rule of thumb, e.g.
```

Comment:
> TODO: for Terra, somewhere on this cheatsheet we need to add [the instruction](https://github.com/verily-src/terra-solutions-team/blob/main/dev_tool_configs/README.md) to install the global gitignore.

It might be better to put these instructions in one of the larger docs (probably part II..) and then link to them.

Reply:
It turns out the info already _was_ in part II (haha) and so I've linked to it.  (I added one bit to those part II instructions, which was to set `export PATH=$PATH:/home/jupyter/.local/bin` if necessary.  My tests show that this path is set for the default gatk VM but not in the Hail cluster terminal.)